specification/cdn/resource-manager/Microsoft.Cdn/preview/2019-06-15-preview/cdnwebapplicationfirewall.json

```diff
@@ -0,0 +1,919 @@
+            "GreaterThanOrEqual",
+            "BeginsWith",
+            "EndsWith",
+            "RegEx"
+          ],
+          "x-ms-enum": {
+            "name": "operator",
+            "modelAsString": true
+          }
+        },
+        "negateCondition": {
+          "description": "Describes if the result of this condition should be negated.",
+          "type": "boolean"
+        },
+        "matchValue": {
+          "description": "List of possible match values.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "ManagedRuleSetList": {
+      "description": "Defines the list of managed rule sets for the policy.",
+      "properties": {
+        "managedRuleSets": {
+          "description": "List of rule sets.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedRuleSet"
+          }
+        }
+      }
+    },
+    "ManagedRuleSet": {
+      "type": "object",
+      "description": "Defines a managed rule set.",
+      "required": [
+        "ruleSetType",
+        "ruleSetVersion"
+      ],
+      "properties": {
+        "ruleSetType": {
+          "description": "Defines the rule set type to use.",
+          "type": "string"
+        },
+        "ruleSetVersion": {
+          "description": "Defines the version of the rule set to use.",
+          "type": "string"
+        },
+        "anomalyScore": {
+          "description": "Verizon only : If the rule set supports anomaly detection mode, this describes the threshold for blocking requests.",
+          "type": "integer",
+          "maximum": 20,
+          "exclusiveMaximum": false,
+          "minimum": 0,
+          "exclusiveMinimum": false
+        },
+        "ruleGroupOverrides": {
+          "description": "Defines the rule overrides to apply to the rule set.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedRuleGroupOverride"
+          }
+        }
+      }
+    },
+    "ManagedRuleGroupOverride": {
+      "description": "Defines a managed rule group override setting.",
+      "required": [
+        "ruleGroupName"
+      ],
+      "properties": {
+        "ruleGroupName": {
+          "description": "Describes the managed rule group within the rule set to override",
+          "type": "string"
+        },
+        "rules": {
+          "description": "List of rules that will be disabled. If none specified, all rules in the group will be disabled.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedRuleOverride"
+          }
+        }
+      }
+    },
+    "ManagedRuleOverride": {
+      "description": "Defines a managed rule group override setting.",
+      "required": [
+        "ruleId"
+      ],
+      "properties": {
+        "ruleId": {
+          "description": "Identifier for the managed rule.",
+          "type": "string"
```

Comment:
Does it have any constraints?

Reply:
this will be discovered via the GET managed rule api and the same name is expected to be used here. In the future when we release the "Stable" version, we will do some work to convert this into an enum with fixed values, but for the preview version, since new managedRuleSets come up  often we do not want to create barriers for their adoption.
The check for unsupported ruleSets for the "preview" version will be on the RP service.

Reply:
ok for preview